wallet/wallet.go

```diff
@@ -363,12 +363,11 @@ func (w *Wallet) syncWithChain(birthdayStamp *waddrmgr.BlockStamp) error {
 		if err != nil {
 			return err
 		}
-		startHeight := bestHeight - waddrmgr.MaxReorgDepth + 1
+		startHeight := birthdayStamp.Height
```

Comment:
We also don't need to check that the birthday height is below 0, since the lowest it can be is 0, i.e., the genesis block.

Reply:
Nit: no need to fetch the hash and header for this again since we already did above.